tests/legacy-cli/e2e/tests/build/server-rendering/server-routes-output-mode-server-platform-neutral.ts

```diff
@@ -0,0 +1,133 @@
+      import { SsrComponent } from './ssr/ssr.component';
+      import { SsgWithParamsComponent } from './ssg-with-params/ssg-with-params.component';
+
+      export const routes: Routes = [
+        {
+          path: '',
+          component: HomeComponent,
+        },
+        {
+          path: 'ssr',
+          component: SsrComponent,
+        },
+        {
+          path: 'ssg/:id',
+          component: SsgWithParamsComponent,
+        },
+      ];
+    `,
+    'src/app/app.routes.server.ts': `
+      import { RenderMode, ServerRoute } from '@angular/ssr';
+
+      export const serverRoutes: ServerRoute[] = [
+        {
+          path: 'ssg/:id',
+          renderMode: RenderMode.Prerender,
+          getPrerenderParams: async() => [{id: 'one'}, {id: 'two'}],
+        },
+        {
+          path: 'ssr',
+          renderMode: RenderMode.Server,
+        },
+        {
+          path: '**',
+          renderMode: RenderMode.Prerender,
+        },
+      ];
+    `,
+    'src/server.ts': `
+      import { AngularAppEngine, createRequestHandler } from '@angular/ssr';
+      import { createApp, createRouter, toWebHandler, defineEventHandler, toWebRequest } from 'h3';
+
+      export const app = createApp();
+      const router = createRouter();
+      const angularAppEngine = new AngularAppEngine();
+
+      router.use(
+        '/**',
+        defineEventHandler((event) => angularAppEngine.render(toWebRequest(event))),
+      );
+
+      app.use(router);
+
+      const handler = toWebHandler(app);
+      export default createRequestHandler(handler);
+    `,
+  });
+  // Generate components for the above routes
+  const componentNames: string[] = ['home', 'ssr', 'ssg-with-params'];
+
+  for (const componentName of componentNames) {
+    await silentNg('generate', 'component', componentName);
+  }
+
+  await updateJsonFile('angular.json', (json) => {
+    const buildTarget = json['projects']['test-project']['architect']['build'];
+    const options = buildTarget['options'];
+    options['ssr']['experimentalPlatform'] = 'neutral';
+    options['outputMode'] = 'server';
+  });
+
+  await noSilentNg('build');
+
+  // Valid SSG pages work
+  const expects: Record<string, string> = {
+    'index.html': 'home works!',
+    'ssg/one/index.html': 'ssg-with-params works!',
+    'ssg/two/index.html': 'ssg-with-params works!',
+  };
+
+  for (const [filePath, fileMatch] of Object.entries(expects)) {
+    await expectFileToMatch(join('dist/test-project/browser', filePath), fileMatch);
+  }
+
+  const filesDoNotExist: string[] = ['csr/index.html', 'ssr/index.html', 'redirect/index.html'];
+  for (const filePath of filesDoNotExist) {
+    const file = join('dist/test-project/browser', filePath);
+    assert.equal(existsSync(file), false, `Expected '${file}' to not exist.`);
+  }
+
+  const port = await findFreePort();
+  await execAndWaitForOutputToMatch(
+    'npx',
+    ['-y', 'listhen', './dist/test-project/server/server.mjs', `--port=${port}`],
```

Comment:
Question: Should `listhen` be in our `devDependencies`? That way our `bazel test` doesn't need network access to download it dynamically and pin the dependency insulate us from breaking changes.

Reply:
Since it’s used only in a single test I preferred to use npx. I can add a version specifier though.